wpilibc/Athena/src/PIDController.cpp

```diff
@@ -338,21 +338,18 @@ void PIDController::SetOutputRange(float minimumOutput, float maximumOutput) {
 void PIDController::SetSetpoint(float setpoint) {
   {
     std::lock_guard<priority_recursive_mutex> sync(m_mutex);
-
     if (m_maximumInput > m_minimumInput) {
       if (setpoint > m_maximumInput)
-        m_setpoint = m_maximumInput;
+        setpoint = m_maximumInput;
       else if (setpoint < m_minimumInput)
-        m_setpoint = m_minimumInput;
-      else
-        m_setpoint = setpoint;
-    } else {
-      m_setpoint = setpoint;
+        setpoint = m_minimumInput;
+    }
+    if(!(m_setpoint==setpoint)){
```

Comment:
So last time we never figured out what to do with this. What if we use the tolerance? That's set by users and has a default value I believe. We should just check that here. So I'd it falls out of tolerance we clear the buffer